uamqp/authentication/cbs_auth.py

```diff
@@ -137,7 +139,14 @@ def handle_token(self):
                 _logger.info("Token on connection %r will expire soon - attempting to refresh.",
                              self._connection.container_id)
                 self.update_token()
-                self._cbs_auth.refresh(self.token, int(self.expires_at))
+                if self.token != self._prev_token:
+                    self._cbs_auth.refresh(self.token, int(self.expires_at))
+                else:
+                    _logger.info(
+                        "The newly acquired token on connection %r is the same as the previous one,"
+                        " will keep attempting to refresh",
+                        self._connection.container_id
```

Comment:
Is this log statement going to get very noisy? Or is it very rare that token == previous_token after calling `update_token`?

Reply:
yes, it would be very noisy in the case `token == previous_token`.

This happens when there's a gap between the client refresh window (uAMQP 6 mins to refresh) and service refresh window (AAD service 5 mins refresh).

So once we have fixed the issue in uamqp/eh/sb, changing default refresh window to 5 mins, it should be rare.

-------------

I think what uamqp is doing technically is not wrong -- it's doing its best to refresh token, however it's doing a bit excessively, retry without backoff. So I think ultimately we might want to introduce kinda backoff to token refresh to let it not throttle